crates/wast/src/wast.rs

```diff
@@ -184,322 +191,251 @@ impl WastContext {
+            }
+            AssertReturnArithmeticNanF64x2 { span: _, invoke } => {
+                for v in self.perform_invoke(invoke)?.into_result()? {
+                    let val = match v {
+                        Val::V128(x) => x,
+                        other => bail!("expected v128, got {:?}", other),
                     };
-                    let bytes = module.encode().map_err(adjust_wast)?;
-                    let err = match self.module(None, &bytes) {
-                        Ok(()) => {
-                            bail!("{}\nexpected module to fail to instantiate", context(span))
+                    for l in 0..4 {
+                        if !is_arithmetic_f64_nan(extract_lane_as_u64(val, l)?) {
+                            bail!("expected f64x2 arithmetic NaN in lane {}", l)
                         }
-                        Err(e) => e,
-                    };
-                    let error_message = format!("{:?}", err);
-                    if !error_message.contains(&message) {
-                        // TODO: change to bail!
-                        println!(
-                            "{}\nassert_malformed: expected {}, got {}",
-                            context(span),
-                            message,
-                            error_message
-                        )
                     }
                 }
-                AssertUnlinkable {
-                    span,
-                    mut module,
-                    message,
-                } => {
-                    let bytes = module.encode().map_err(adjust_wast)?;
-                    let err = match self.module(None, &bytes) {
-                        Ok(()) => bail!("{}\nexpected module to fail to link", context(span)),
-                        Err(e) => e,
-                    };
-                    let error_message = format!("{:?}", err);
-                    if !error_message.contains(&message) {
-                        bail!(
-                            "{}\nassert_unlinkable: expected {}, got {}",
-                            context(span),
-                            message,
-                            error_message
-                        )
-                    }
+            }
+            AssertInvalid {
+                span: _,
+                mut module,
+                message,
+            } => {
+                let bytes = module.encode()?;
+                let err = match self.module(None, &bytes) {
+                    Ok(()) => bail!("expected module to fail to build"),
+                    Err(e) => e,
+                };
+                let error_message = format!("{:?}", err);
+                if !error_message.contains(&message) {
+                    // TODO: change to bail!
+                    println!(
+                        "assert_invalid: expected {}, got {}",
+                        message, error_message
+                    )
```

Comment:
TODO: why can't we bail here?

Reply:
I suspect that this is because wasmtime's error messages don't always match the error expectations, and it's probably not worth a ton of effort to get the error messages to all line up. FWIW though this is [just carrying over an existing TODO](https://github.com/bytecodealliance/wasmtime/blob/aa41d4b12bfb18ec34fecfde0f37a426674c63e3/crates/wast/src/wast.rs#L443-L449) rather than adding a new one.